discojs/src/aggregator/mean.spec.ts

```diff
@@ -11,44 +11,115 @@ async function WSIntoArrays(ws: WeightsContainer): Promise<number[][]> {
       aggregator.once("aggregation", resolve),
     );
 
-    aggregator.add("first client", WeightsContainer.of([0], [1]), 0);
-    aggregator.add("second client", WeightsContainer.of([2], [3]), 0);
+    const result1 = aggregator.add(id1, WeightsContainer.of([0], [1]));
+    const result2 = aggregator.add(id2, WeightsContainer.of([2], [3]));
+    expect((await result1).equals(await result2)).to.be.true
 
     expect(await WSIntoArrays(await results)).to.deep.equal([[1], [2]]);
   });
+
+  it("waits for 100% of the contributions by default", async () => {
+    const aggregator = new MeanAggregator();
+    const [id1, id2] = ["client 1", "client 2"]
+
+    aggregator.setNodes(Set.of(id1, id2));
+
+    const result1 = aggregator.add(id1, WeightsContainer.of([0], [1]));
+    // Make sure that the aggregation isn't triggered by waiting some time
+    let timeoutTriggered = false
+    await Promise.race([result1,
+      new Promise<void>(resolve => setTimeout(() => {
+        timeoutTriggered = true
+        resolve()
+      }, 300))
+    ])
+    expect(timeoutTriggered).to.be.true
+    
+    aggregator.registerNode(id2);
+    const result2 = aggregator.add(id2, WeightsContainer.of([2], [3]));
+    expect((await result1).equals(await result2)).to.be.true
+  });
+
+  it("can wait for an absolute number of contributions", async () => {
+    const aggregator = new MeanAggregator(0, 1, 'absolute');
+    const [id1, id2] = ["client 1", "client 2"]
+    aggregator.setNodes(Set.of(id1, id2)); // register two clients 
+
+    // should aggregate with only one contribution
+    const result = await aggregator.add(id1, WeightsContainer.of([0], [1]));
+    expect(await WSIntoArrays(result)).to.deep.equal([[0], [1]]);
+  });
+
+  it("can wait for an relative number of contributions", async () => {
+    const aggregator = new MeanAggregator(0, 0.5, 'relative');
+    const [id1, id2] = ["client 1", "client 2"]
+    aggregator.setNodes(Set.of(id1, id2)); // register two clients 
+
+    // should aggregate with only 50% of the contribution (1 contribution)
+    const result = await aggregator.add(id1, WeightsContainer.of([0], [1]));
+    expect(await WSIntoArrays(result)).to.deep.equal([[0], [1]]);
+  });
+  
+  it("doesn't aggregate when not enough participants", async () => {
+    const aggregator = new MeanAggregator(0, 1, 'absolute'); // only wait for a single participant
+    aggregator.minNbOfParticipants = 2 // However the task can specify another minimum number, here 2
+    const [id1, id2] = ["client 1", "client 2"]
+    aggregator.setNodes(Set.of(id1));
+    
+    const result1 = aggregator.add(id1, WeightsContainer.of([0], [1]));
+    // Make sure that the aggregation isn't triggered by waiting some time
+    let timeoutTriggered = false
+    await Promise.race([result1,
+      new Promise<void>(resolve => setTimeout(() => {
+        timeoutTriggered = true
+        resolve()
+      }, 300))
+    ])
+    expect(timeoutTriggered).to.be.true
```

Comment:
time based tests, same fix idea

(and I played with it a bit, you can avoid the `let` with that)

```suggestion
    const timeoutTriggered = await Promise.race([
      result1,
      new Promise<void>((resolve) => setTimeout(resolve, 300)),
    ]);
    expect(timeoutTriggered).to.be.undefined;
```